internal/commands/app/initialize/command.go

```diff
@@ -0,0 +1,97 @@
+package initialize
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+
+	"github.com/spf13/pflag"
+)
+
+// Command is the `app init` command
+var Command = cli.CommandDefinition{
+	Use:         "init",
+	Aliases:     []string{"initialize"},
+	Display:     "app init",
+	Description: "Initialize a Realm app in your current local directory",
+	Help:        "",
+	Command:     &command{},
+}
+
+type command struct {
+	inputs      inputs
+	realmClient realm.Client
+}
+
+func (cmd *command) Flags(fs *pflag.FlagSet) {
+	fs.StringVar(&cmd.inputs.Project, flagProject, "", flagProjectUsage)
+	fs.StringVarP(&cmd.inputs.From, flagFrom, flagFromShort, "", flagFromUsage)
+	fs.StringVarP(&cmd.inputs.Name, flagName, flagNameShort, "", flagNameUsage)
+	fs.VarP(&cmd.inputs.DeploymentModel, flagDeploymentModel, flagDeploymentModelShort, flagDeploymentModelUsage)
+	fs.VarP(&cmd.inputs.Location, flagLocation, flagLocationShort, flagLocationUsage)
+}
+
+func (cmd *command) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+func (cmd *command) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = realm.NewAuthClient(profile.RealmBaseURL(), profile.Session())
+	return nil
+}
+
+func (cmd *command) Handler(profile *cli.Profile, ui terminal.UI) error {
+	from, fromErr := cmd.inputs.resolveFrom(ui, cmd.realmClient)
+	if fromErr != nil {
+		return fromErr
+	}
+
+	switch from.Type {
+	case fromApp:
+		return cmd.initializeFromApp(profile.WorkingDirectory, from.GroupID, from.AppID)
+	case fromTemplate:
+		return cmd.initializeFromTemplate(profile.WorkingDirectory, from.TemplateID)
+	}
+	return cmd.initialize(profile.WorkingDirectory)
+}
+
+func (cmd *command) Feedback(profile *cli.Profile, ui terminal.UI) error {
+	return ui.Print(terminal.NewTextLog("Successfully initialized app"))
+}
+
+func (cmd *command) initialize(wd string) error {
+	data := []byte(fmt.Sprintf(`{
```

Comment:
what was the reasoning here between using a struct / json marshalling vs doing this? I feel like a struct is less likely for us to make a mistake typing this out.

Reply:
simplicity? no real reasoning...we don't rely on knowing the structure anywhere else so I just went for the string payload here.  In general, we won't care much about a lot of the various file contents/structures, though I can see the argument being made fairly quickly that we do care at least for this app/root `config.json` while we attempt to pull app information out of it (id, name) and/or merge data into it (when creating from template or existing app).

i'll add it as a struct with json tags and work with it from that perspective